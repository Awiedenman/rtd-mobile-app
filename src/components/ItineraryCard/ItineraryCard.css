.itinerary-card {
  height: 20%;
  width: 100%;
  background: white;
  box-shadow: 1px 1px 2px 2px rgba(0, 0, 0, 0.5);
}

.top-container {
  height: 40px;
  display: flex;
  align-content: flex-end;
  padding: 5px;
}

.transit-icon {
  height: 30px;
  width: auto;
}

.line-id-container {
  height: 30px;
  width: 30px;
  display: flex;
  justify-content: center;
  align-items: center;
  border-radius: 2px;
  margin: 0 5px 0 5px;
}

.add-favorite {
  border: 1px solid black;
}

.line-id {
  color: white;
}

.trip-detail-row {
  display: grid;
  grid-template-columns: 20% 1fr 20%;
  align-items: center;
  justify-items: center;
}

.dot {
  height: 10px;
  width: 10px;
  background: black;
  border-radius: 50%;
}

.duration-line {
  grid-column: 2 / 3;
  border: 1px solid black;
}

.bottom-container {
  display: flex;
  justify-content: space-between;
  padding-right: 10px;
}