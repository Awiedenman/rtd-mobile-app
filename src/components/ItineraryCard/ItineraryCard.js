import React, { Component } from 'react';
import PropTypes from 'prop-types';
import { infoCleaner, cleanStep } from '../../constants/cleanerFunctions';
import ItineraryStep from '../ItineraryStep/ItineraryStep';

import './ItineraryCard.css';

export class ItineraryCard extends Component {
  constructor(props){
    super(props);
    this.state = {
      showMore: false
    };
  } 

  handleClick = () => {
    this.setState({
      showMore: !this.state.showMore
    });
  }

  addressCleaner = (address) => {
    return address.split(',')[0];
  }

  render(){
    
    const {
      itinerary_id,
      arrival_time,
      duration,
      steps
    } = this.props.itinerary;

    const info = infoCleaner(steps);

    const lineStyles = {
      background: info.color
    };

    if (this.state.showMore) {
      return steps.map((step, index) => {
        const props = cleanStep(step);
        return (
          <ItineraryStep 
            data={props}
            key={index}
            handleClick={this.handleClick}
          />
        );
      });
    } else {
      return (
        <div 
          className='itinerary-card'
          id ={itinerary_id}
          onClick={this.handleClick}
        >
          <div className='top-container'>
            <img
              alt='transit-icon'
              className='transit-icon'
              src={info.imgUrl} 
            />
            <div 
              className='line-id-container'
              style={lineStyles}
            >
              <p className='line-id'>
                {info.line}
              </p>
            </div>
            <h2>
              Towards {info.headsign}
            </h2>
          </div>
          <div className='trip-detail-row'>
            <h3 className="time-depart">
              {info.departure_time}
            </h3>
            <div className='duration-line-container' >
              <span className='dot'></span>
              <hr className='duration-line' />
            </div>
            <h3 className="time-depart">
              {arrival_time}
            </h3>
          </div>
          <div className='bottom-container'>
            <h2>
              From {info.departure_stop}
            </h2>
            <h2>{duration}</h2>
          </div>
        </div>
      );
    }
  }
}

export default ItineraryCard;

ItineraryCard.propTypes = {
  itinerary: PropTypes.object,
  addFavorite: PropTypes.func,
  uid: PropTypes.string
};