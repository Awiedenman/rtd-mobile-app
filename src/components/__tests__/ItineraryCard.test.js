import React from 'react';
import { shallow } from 'enzyme';
import { ItineraryCard } from '../ItineraryCard/ItineraryCard';

describe('ItineraryCard', () => {
  let wrapper;
  let mockProps;
  beforeEach(() => {
    mockProps = {
      key: 1,
    itinerary: { 
      arrival_time:"6:33am",
      departure_time:"5:51am",
      distance:"8.1 mi",
      duration:"42 mins",
      end_address:"Columbine St, Denver, CO, USA",
      favorite:false,
      itinerary_id:59,
      start_address:"S Monaco Pkwy, Colorado, USA",
      steps: [{
        arrival_stop: "Monaco Pkwy & 29th Ave",
        arrival_time:"6:07am",
        color:"#0277bd",
        credit_name:"Regional Transportation District",
        credit_url:"http://rtd-denver.com/",
        departure_stop:"S Monaco Pkwy & Evans Ave",
        departure_time:"5:51am",
        distance:"5.5 mi",
        duration:"16 mins",
        headsign:"Central Park 65 via Monaco",
        id:656,
        instructions:"Bus towards Central Park 65 via Monaco",
        name:"Monaco Parkway",
        num_stops:24,
        possible_route_id:169,
        short_name:"65",
        travel_mode:"TRANSIT",
        vehicle_type:"Bus",
       } ],
      title: null 
    },
    uid: '3',
    addFavorite: jest.fn()
  }
    wrapper = shallow(<ItineraryCard {...mockProps} />)
  })
  it('should match snapshot', () => {
    
    expect(wrapper).toMatchSnapshot();
  })

  it('should match snapshot if a favorite is truthy', () => {
    
    wrapper.setState({favorite: true})

    expect(wrapper).toMatchSnapshot();
  })
   
  describe('handleClick', () => {
    it('should toggle the showMore state', () => {
      const expected= true;

      wrapper.instance().handleClick();

      expect(wrapper.state('showMore')).toEqual(expected);
    })
  })

  describe('addressCleaner', () => {
    test('should clean address', () => {
      const mockAddress = 'Union Station, Denver,CO';
      const expected = 'Union Station';
      const result = wrapper.instance().addressCleaner(mockAddress);
  
      expect(result).toEqual(expected);
    })
  })
  
  
  
})
