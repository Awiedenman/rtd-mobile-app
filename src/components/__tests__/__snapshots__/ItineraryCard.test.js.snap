// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ItineraryCard should match snapshot 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <ItineraryCard
    addFavorite={[Function]}
    itinerary={
        Object {
            "arrival_time": "6:33am",
            "departure_time": "5:51am",
            "distance": "8.1 mi",
            "duration": "42 mins",
            "end_address": "Columbine St, Denver, CO, USA",
            "favorite": false,
            "itinerary_id": 59,
            "start_address": "S Monaco Pkwy, Colorado, USA",
            "steps": Array [
              Object {
                "arrival_stop": "Monaco Pkwy & 29th Ave",
                "arrival_time": "6:07am",
                "color": "#0277bd",
                "credit_name": "Regional Transportation District",
                "credit_url": "http://rtd-denver.com/",
                "departure_stop": "S Monaco Pkwy & Evans Ave",
                "departure_time": "5:51am",
                "distance": "5.5 mi",
                "duration": "16 mins",
                "headsign": "Central Park 65 via Monaco",
                "id": 656,
                "instructions": "Bus towards Central Park 65 via Monaco",
                "name": "Monaco Parkway",
                "num_stops": 24,
                "possible_route_id": 169,
                "short_name": "65",
                "travel_mode": "TRANSIT",
                "vehicle_type": "Bus",
              },
            ],
            "title": null,
          }
    }
    uid="3"
/>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <div
          className="top-container"
>
          <img
                    alt="transit-icon"
                    className="transit-icon"
                    src="bus.png"
          />
          <div
                    className="line-id-container"
                    style={
                              Object {
                                        "background": "#0277bd",
                                      }
                    }
          >
                    <p
                              className="line-id"
                    >
                              65
                    </p>
          </div>
          <h2>
                    Towards 
                    Central Park 65 via Monaco
          </h2>
</div>,
        <div
          className="trip-detail-row"
>
          <h3
                    className="time-depart"
          >
                    5:51am
          </h3>
          <div
                    className="duration-line-container"
          >
                    <span
                              className="dot"
                    />
                    <hr
                              className="duration-line"
                    />
          </div>
          <h3
                    className="time-depart"
          >
                    6:33am
          </h3>
</div>,
        <div
          className="bottom-container"
>
          <h2>
                    From 
                    S Monaco Pkwy & Evans Ave
          </h2>
          <h2>
                    42 mins
          </h2>
</div>,
      ],
      "className": "itinerary-card",
      "id": undefined,
      "onClick": [Function],
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            <img
              alt="transit-icon"
              className="transit-icon"
              src="bus.png"
/>,
            <div
              className="line-id-container"
              style={
                            Object {
                                          "background": "#0277bd",
                                        }
              }
>
              <p
                            className="line-id"
              >
                            65
              </p>
</div>,
            <h2>
              Towards 
              Central Park 65 via Monaco
</h2>,
          ],
          "className": "top-container",
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "alt": "transit-icon",
              "className": "transit-icon",
              "src": "bus.png",
            },
            "ref": null,
            "rendered": null,
            "type": "img",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": <p
                className="line-id"
>
                65
</p>,
              "className": "line-id-container",
              "style": Object {
                "background": "#0277bd",
              },
            },
            "ref": null,
            "rendered": Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": "65",
                "className": "line-id",
              },
              "ref": null,
              "rendered": "65",
              "type": "p",
            },
            "type": "div",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": Array [
                "Towards ",
                "Central Park 65 via Monaco",
              ],
            },
            "ref": null,
            "rendered": Array [
              "Towards ",
              "Central Park 65 via Monaco",
            ],
            "type": "h2",
          },
        ],
        "type": "div",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            <h3
              className="time-depart"
>
              5:51am
</h3>,
            <div
              className="duration-line-container"
>
              <span
                            className="dot"
              />
              <hr
                            className="duration-line"
              />
</div>,
            <h3
              className="time-depart"
>
              6:33am
</h3>,
          ],
          "className": "trip-detail-row",
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": "5:51am",
              "className": "time-depart",
            },
            "ref": null,
            "rendered": "5:51am",
            "type": "h3",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": Array [
                <span
                  className="dot"
/>,
                <hr
                  className="duration-line"
/>,
              ],
              "className": "duration-line-container",
            },
            "ref": null,
            "rendered": Array [
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "className": "dot",
                },
                "ref": null,
                "rendered": null,
                "type": "span",
              },
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "className": "duration-line",
                },
                "ref": null,
                "rendered": null,
                "type": "hr",
              },
            ],
            "type": "div",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": "6:33am",
              "className": "time-depart",
            },
            "ref": null,
            "rendered": "6:33am",
            "type": "h3",
          },
        ],
        "type": "div",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            <h2>
              From 
              S Monaco Pkwy & Evans Ave
</h2>,
            <h2>
              42 mins
</h2>,
          ],
          "className": "bottom-container",
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": Array [
                "From ",
                "S Monaco Pkwy & Evans Ave",
              ],
            },
            "ref": null,
            "rendered": Array [
              "From ",
              "S Monaco Pkwy & Evans Ave",
            ],
            "type": "h2",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": "42 mins",
            },
            "ref": null,
            "rendered": "42 mins",
            "type": "h2",
          },
        ],
        "type": "div",
      },
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <div
            className="top-container"
>
            <img
                        alt="transit-icon"
                        className="transit-icon"
                        src="bus.png"
            />
            <div
                        className="line-id-container"
                        style={
                                    Object {
                                                "background": "#0277bd",
                                              }
                        }
            >
                        <p
                                    className="line-id"
                        >
                                    65
                        </p>
            </div>
            <h2>
                        Towards 
                        Central Park 65 via Monaco
            </h2>
</div>,
          <div
            className="trip-detail-row"
>
            <h3
                        className="time-depart"
            >
                        5:51am
            </h3>
            <div
                        className="duration-line-container"
            >
                        <span
                                    className="dot"
                        />
                        <hr
                                    className="duration-line"
                        />
            </div>
            <h3
                        className="time-depart"
            >
                        6:33am
            </h3>
</div>,
          <div
            className="bottom-container"
>
            <h2>
                        From 
                        S Monaco Pkwy & Evans Ave
            </h2>
            <h2>
                        42 mins
            </h2>
</div>,
        ],
        "className": "itinerary-card",
        "id": undefined,
        "onClick": [Function],
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              <img
                alt="transit-icon"
                className="transit-icon"
                src="bus.png"
/>,
              <div
                className="line-id-container"
                style={
                                Object {
                                                "background": "#0277bd",
                                              }
                }
>
                <p
                                className="line-id"
                >
                                65
                </p>
</div>,
              <h2>
                Towards 
                Central Park 65 via Monaco
</h2>,
            ],
            "className": "top-container",
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "alt": "transit-icon",
                "className": "transit-icon",
                "src": "bus.png",
              },
              "ref": null,
              "rendered": null,
              "type": "img",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": <p
                  className="line-id"
>
                  65
</p>,
                "className": "line-id-container",
                "style": Object {
                  "background": "#0277bd",
                },
              },
              "ref": null,
              "rendered": Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": "65",
                  "className": "line-id",
                },
                "ref": null,
                "rendered": "65",
                "type": "p",
              },
              "type": "div",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": Array [
                  "Towards ",
                  "Central Park 65 via Monaco",
                ],
              },
              "ref": null,
              "rendered": Array [
                "Towards ",
                "Central Park 65 via Monaco",
              ],
              "type": "h2",
            },
          ],
          "type": "div",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              <h3
                className="time-depart"
>
                5:51am
</h3>,
              <div
                className="duration-line-container"
>
                <span
                                className="dot"
                />
                <hr
                                className="duration-line"
                />
</div>,
              <h3
                className="time-depart"
>
                6:33am
</h3>,
            ],
            "className": "trip-detail-row",
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": "5:51am",
                "className": "time-depart",
              },
              "ref": null,
              "rendered": "5:51am",
              "type": "h3",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": Array [
                  <span
                    className="dot"
/>,
                  <hr
                    className="duration-line"
/>,
                ],
                "className": "duration-line-container",
              },
              "ref": null,
              "rendered": Array [
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "host",
                  "props": Object {
                    "className": "dot",
                  },
                  "ref": null,
                  "rendered": null,
                  "type": "span",
                },
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "host",
                  "props": Object {
                    "className": "duration-line",
                  },
                  "ref": null,
                  "rendered": null,
                  "type": "hr",
                },
              ],
              "type": "div",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": "6:33am",
                "className": "time-depart",
              },
              "ref": null,
              "rendered": "6:33am",
              "type": "h3",
            },
          ],
          "type": "div",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              <h2>
                From 
                S Monaco Pkwy & Evans Ave
</h2>,
              <h2>
                42 mins
</h2>,
            ],
            "className": "bottom-container",
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": Array [
                  "From ",
                  "S Monaco Pkwy & Evans Ave",
                ],
              },
              "ref": null,
              "rendered": Array [
                "From ",
                "S Monaco Pkwy & Evans Ave",
              ],
              "type": "h2",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": "42 mins",
              },
              "ref": null,
              "rendered": "42 mins",
              "type": "h2",
            },
          ],
          "type": "div",
        },
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
    "attachTo": undefined,
    "hydrateIn": undefined,
  },
}
`;

exports[`ItineraryCard should match snapshot if a favorite is truthy 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <ItineraryCard
    addFavorite={[Function]}
    itinerary={
        Object {
            "arrival_time": "6:33am",
            "departure_time": "5:51am",
            "distance": "8.1 mi",
            "duration": "42 mins",
            "end_address": "Columbine St, Denver, CO, USA",
            "favorite": false,
            "itinerary_id": 59,
            "start_address": "S Monaco Pkwy, Colorado, USA",
            "steps": Array [
              Object {
                "arrival_stop": "Monaco Pkwy & 29th Ave",
                "arrival_time": "6:07am",
                "color": "#0277bd",
                "credit_name": "Regional Transportation District",
                "credit_url": "http://rtd-denver.com/",
                "departure_stop": "S Monaco Pkwy & Evans Ave",
                "departure_time": "5:51am",
                "distance": "5.5 mi",
                "duration": "16 mins",
                "headsign": "Central Park 65 via Monaco",
                "id": 656,
                "instructions": "Bus towards Central Park 65 via Monaco",
                "name": "Monaco Parkway",
                "num_stops": 24,
                "possible_route_id": 169,
                "short_name": "65",
                "travel_mode": "TRANSIT",
                "vehicle_type": "Bus",
              },
            ],
            "title": null,
          }
    }
    uid="3"
/>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <div
          className="top-container"
>
          <img
                    alt="transit-icon"
                    className="transit-icon"
                    src="bus.png"
          />
          <div
                    className="line-id-container"
                    style={
                              Object {
                                        "background": "#0277bd",
                                      }
                    }
          >
                    <p
                              className="line-id"
                    >
                              65
                    </p>
          </div>
          <h2>
                    Towards 
                    Central Park 65 via Monaco
          </h2>
</div>,
        <div
          className="trip-detail-row"
>
          <h3
                    className="time-depart"
          >
                    5:51am
          </h3>
          <div
                    className="duration-line-container"
          >
                    <span
                              className="dot"
                    />
                    <hr
                              className="duration-line"
                    />
          </div>
          <h3
                    className="time-depart"
          >
                    6:33am
          </h3>
</div>,
        <div
          className="bottom-container"
>
          <h2>
                    From 
                    S Monaco Pkwy & Evans Ave
          </h2>
          <h2>
                    42 mins
          </h2>
</div>,
      ],
      "className": "itinerary-card",
      "id": undefined,
      "onClick": [Function],
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            <img
              alt="transit-icon"
              className="transit-icon"
              src="bus.png"
/>,
            <div
              className="line-id-container"
              style={
                            Object {
                                          "background": "#0277bd",
                                        }
              }
>
              <p
                            className="line-id"
              >
                            65
              </p>
</div>,
            <h2>
              Towards 
              Central Park 65 via Monaco
</h2>,
          ],
          "className": "top-container",
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "alt": "transit-icon",
              "className": "transit-icon",
              "src": "bus.png",
            },
            "ref": null,
            "rendered": null,
            "type": "img",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": <p
                className="line-id"
>
                65
</p>,
              "className": "line-id-container",
              "style": Object {
                "background": "#0277bd",
              },
            },
            "ref": null,
            "rendered": Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": "65",
                "className": "line-id",
              },
              "ref": null,
              "rendered": "65",
              "type": "p",
            },
            "type": "div",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": Array [
                "Towards ",
                "Central Park 65 via Monaco",
              ],
            },
            "ref": null,
            "rendered": Array [
              "Towards ",
              "Central Park 65 via Monaco",
            ],
            "type": "h2",
          },
        ],
        "type": "div",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            <h3
              className="time-depart"
>
              5:51am
</h3>,
            <div
              className="duration-line-container"
>
              <span
                            className="dot"
              />
              <hr
                            className="duration-line"
              />
</div>,
            <h3
              className="time-depart"
>
              6:33am
</h3>,
          ],
          "className": "trip-detail-row",
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": "5:51am",
              "className": "time-depart",
            },
            "ref": null,
            "rendered": "5:51am",
            "type": "h3",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": Array [
                <span
                  className="dot"
/>,
                <hr
                  className="duration-line"
/>,
              ],
              "className": "duration-line-container",
            },
            "ref": null,
            "rendered": Array [
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "className": "dot",
                },
                "ref": null,
                "rendered": null,
                "type": "span",
              },
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "className": "duration-line",
                },
                "ref": null,
                "rendered": null,
                "type": "hr",
              },
            ],
            "type": "div",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": "6:33am",
              "className": "time-depart",
            },
            "ref": null,
            "rendered": "6:33am",
            "type": "h3",
          },
        ],
        "type": "div",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            <h2>
              From 
              S Monaco Pkwy & Evans Ave
</h2>,
            <h2>
              42 mins
</h2>,
          ],
          "className": "bottom-container",
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": Array [
                "From ",
                "S Monaco Pkwy & Evans Ave",
              ],
            },
            "ref": null,
            "rendered": Array [
              "From ",
              "S Monaco Pkwy & Evans Ave",
            ],
            "type": "h2",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": "42 mins",
            },
            "ref": null,
            "rendered": "42 mins",
            "type": "h2",
          },
        ],
        "type": "div",
      },
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <div
            className="top-container"
>
            <img
                        alt="transit-icon"
                        className="transit-icon"
                        src="bus.png"
            />
            <div
                        className="line-id-container"
                        style={
                                    Object {
                                                "background": "#0277bd",
                                              }
                        }
            >
                        <p
                                    className="line-id"
                        >
                                    65
                        </p>
            </div>
            <h2>
                        Towards 
                        Central Park 65 via Monaco
            </h2>
</div>,
          <div
            className="trip-detail-row"
>
            <h3
                        className="time-depart"
            >
                        5:51am
            </h3>
            <div
                        className="duration-line-container"
            >
                        <span
                                    className="dot"
                        />
                        <hr
                                    className="duration-line"
                        />
            </div>
            <h3
                        className="time-depart"
            >
                        6:33am
            </h3>
</div>,
          <div
            className="bottom-container"
>
            <h2>
                        From 
                        S Monaco Pkwy & Evans Ave
            </h2>
            <h2>
                        42 mins
            </h2>
</div>,
        ],
        "className": "itinerary-card",
        "id": undefined,
        "onClick": [Function],
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              <img
                alt="transit-icon"
                className="transit-icon"
                src="bus.png"
/>,
              <div
                className="line-id-container"
                style={
                                Object {
                                                "background": "#0277bd",
                                              }
                }
>
                <p
                                className="line-id"
                >
                                65
                </p>
</div>,
              <h2>
                Towards 
                Central Park 65 via Monaco
</h2>,
            ],
            "className": "top-container",
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "alt": "transit-icon",
                "className": "transit-icon",
                "src": "bus.png",
              },
              "ref": null,
              "rendered": null,
              "type": "img",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": <p
                  className="line-id"
>
                  65
</p>,
                "className": "line-id-container",
                "style": Object {
                  "background": "#0277bd",
                },
              },
              "ref": null,
              "rendered": Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": "65",
                  "className": "line-id",
                },
                "ref": null,
                "rendered": "65",
                "type": "p",
              },
              "type": "div",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": Array [
                  "Towards ",
                  "Central Park 65 via Monaco",
                ],
              },
              "ref": null,
              "rendered": Array [
                "Towards ",
                "Central Park 65 via Monaco",
              ],
              "type": "h2",
            },
          ],
          "type": "div",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              <h3
                className="time-depart"
>
                5:51am
</h3>,
              <div
                className="duration-line-container"
>
                <span
                                className="dot"
                />
                <hr
                                className="duration-line"
                />
</div>,
              <h3
                className="time-depart"
>
                6:33am
</h3>,
            ],
            "className": "trip-detail-row",
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": "5:51am",
                "className": "time-depart",
              },
              "ref": null,
              "rendered": "5:51am",
              "type": "h3",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": Array [
                  <span
                    className="dot"
/>,
                  <hr
                    className="duration-line"
/>,
                ],
                "className": "duration-line-container",
              },
              "ref": null,
              "rendered": Array [
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "host",
                  "props": Object {
                    "className": "dot",
                  },
                  "ref": null,
                  "rendered": null,
                  "type": "span",
                },
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "host",
                  "props": Object {
                    "className": "duration-line",
                  },
                  "ref": null,
                  "rendered": null,
                  "type": "hr",
                },
              ],
              "type": "div",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": "6:33am",
                "className": "time-depart",
              },
              "ref": null,
              "rendered": "6:33am",
              "type": "h3",
            },
          ],
          "type": "div",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              <h2>
                From 
                S Monaco Pkwy & Evans Ave
</h2>,
              <h2>
                42 mins
</h2>,
            ],
            "className": "bottom-container",
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": Array [
                  "From ",
                  "S Monaco Pkwy & Evans Ave",
                ],
              },
              "ref": null,
              "rendered": Array [
                "From ",
                "S Monaco Pkwy & Evans Ave",
              ],
              "type": "h2",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": "42 mins",
              },
              "ref": null,
              "rendered": "42 mins",
              "type": "h2",
            },
          ],
          "type": "div",
        },
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
    "attachTo": undefined,
    "hydrateIn": undefined,
  },
}
`;
